js/models/messages.js

```diff
@@ -2044,33 +2193,43 @@
         this.set({ endSessionType: 'ongoing' });
       }
 
-      if (message.isFriendRequest() && isSessionRequest) {
-        // Check if the contact is a member in one of our private groups:
-        const groupMember = window
-          .getConversations()
-          .models.filter(c => c.get('members'))
-          .reduce((acc, x) => window.Lodash.concat(acc, x.get('members')), [])
-          .includes(primarySource);
-
-        if (groupMember) {
-          window.log.info(
-            `Auto accepting a 'group' friend request for a known group member: ${primarySource}`
-          );
-
-          window.libloki.api.sendBackgroundMessage(message.get('source'));
+      if (message.isSessionRequest()) {
+        await this.handleSessionRequest(source, primarySource, confirm);
 
-          confirm();
-        }
-
-        // Wether or not we accepted the FR, we exit early so background friend requests
+        // Wether or not we accepted the FR, we exit early so session requests
         // cannot be used for establishing regular private conversations
         return null;
       }
 
+      // Session request have been dealt with before, so a friend request here is
+      // not a session request message. Also, handleAutoFriendRequestMessage() only handles the autoAccept logic of an auto friend request.
+      // FIXME audric: is that correct? Also, if you recover your account from mnemonic for instance,
+      // people will know you as a friend but you won't have that information.
+      // so they will send basic messages, but for us, it needs to be handled as an FR.
+      if (
+        message.isFriendRequest() ||
+        (!isGroupMessage && !conversationOrigin.isFriend())
+      ) {
+        const shouldReturn = await this.handleAutoFriendRequestMessage(
+          source,
+          ourNumber,
+          conversationOrigin,
+          confirm
+        );
+        // handleAutoFriendRequestMessage can process fully a message in some cases
+        // so we need to return early if that's the case
+        if (shouldReturn) {
+          return null;
+        }
+      }
+      const conversation = conversationPrimary;
```

Comment:
next process is with the conversation of the primaryDevice because that one was done at before:
https://github.com/loki-project/session-desktop/blob/clearnet/js/models/messages.js#L1920-L1929